### PR TITLE
Merge SkyCoord support to Master

### PR DESCRIPTION
**Pull request by Xen0Xys**
## TODO
- [x] Fix positionChanged event when position.ra/dec is different that getRaDec result
- [x] Fix example 6 bug with position sync between multiple aladin views
- [x] Add SkyCoord when the target is a coordinate string
- [x] Add documentation
- [x] Check that existing examples are always working
- [x] Fix target initialization parameter
- [x] Add testing for parsing methods
- [x] Add changelog

### CODE REVIEW

**Comment by ManonMarchand** [comment]
Thanks Tom!

Fixes: #51 

For the gitignore, you can edit your local `.git/info/exclude` to avoid editing the shared one

I'll have a look at the rest tomorrow :slightly_smiling_face:

File: src/ipyaladin/__init__.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ class Aladin(anywidget.AnyWidget):
 
     # Options for the view initialization
     height = Int(400).tag(sync=True, init_option=True)
-    target = Unicode("0 0").tag(sync=True, init_option=True)
+    _target = Unicode("0 0").tag(sync=True, init_option=True)
+    _frame = Unicode("icrs").tag(sync=True, init_option=True)
```
Comment (ManonMarchand) [inline src/ipyaladin/__init__.py:33]: There is already a coo_frame trait that you might be able to use here

Comment (Xen0Xys) [reply src/ipyaladin/__init__.py:33]: The frame traitlets was used in previous testing, I will remove it

File: src/ipyaladin/__init__.py [hunk 2]
```diff
@@ -105,6 +105,56 @@ def _handle_custom_message(self, model, message, list_of_buffers):  # noqa: ARG0
         elif event_type == "select" and "select" in self.listener_callback:
             self.listener_callback["select"](message_content)
 
+    @property
+    def target(self):
+        from astropy.coordinates import SkyCoord
+
+        try:
+            return SkyCoord(
+                ra=self._target.split(" ")[0],
```
Comment (bmatthieu3) [inline src/ipyaladin/__init__.py:114]: You could do only one self._target.split that does for the ra and the dec.

File: src/ipyaladin/__init__.py [hunk 3]
```diff
@@ -105,6 +105,56 @@ def _handle_custom_message(self, model, message, list_of_buffers):  # noqa: ARG0
...
+                self.send(
+                    {
+                        "event_name": "goto_ra_dec",
+                        "ra": sc.icrs.ra.deg,
+                        "dec": sc.icrs.dec.deg,
+                    }
+                )
+            else:  # If the target is a coordinate string
+                self._target = target
+                self.send({"event_name": "goto_ra_dec", "object": target})
```
Comment (bmatthieu3) [inline src/ipyaladin/__init__.py:142]: I think the event_name should be goto_object here

File: src/ipyaladin/__init__.py [hunk 4]
```diff
@@ -105,6 +105,56 @@ def _handle_custom_message(self, model, message, list_of_buffers):  # noqa: ARG0
...
+                sc = SkyCoord.from_name(target)
+                self._target = f"{sc.icrs.ra.deg} {sc.icrs.dec.deg}"
+                self.send(
+                    {
+                        "event_name": "goto_ra_dec",
+                        "ra": sc.icrs.ra.deg,
+                        "dec": sc.icrs.dec.deg,
+                    }
+                )
+            else:  # If the target is a coordinate string
```
Comment (bmatthieu3) [inline src/ipyaladin/__init__.py:140]: Maybe we could just add a basic test to check if there is at least 2 values separated by a space before sending it to gotoObject of aladin. Aladin Lite does not move if we just give for example:
```python
aladin.target = '14.89'
```

File: src/ipyaladin/__init__.py [hunk 5]
```diff
@@ -105,6 +105,56 @@ def _handle_custom_message(self, model, message, list_of_buffers):  # noqa: ARG0
...
+                    "event_name": "goto_ra_dec",
+                    "ra": target.icrs.ra.deg,
+                    "dec": target.icrs.dec.deg,
+                }
+            )
+        else:
+            raise ValueError(
+                "target must be a string or an astropy.coordinates.SkyCoord object"
+            )
+        self.send({"event_name": "change_target", "target": self._target})
```
Comment (bmatthieu3) [inline src/ipyaladin/__init__.py:156]: Should it not be a goto_ra_dec event as well here ?

**Review by bmatthieu3 — CHANGES_REQUESTED** [review]
Thanks @Xen0Xys . I do have some little things to suggest. Overall it is very good :D

File: js/widget.js [hunk 6]
```diff
@@ -235,7 +251,6 @@ function render({ model, el }) {
 
   return () => {
     // need to unsubscribe the listeners
-    model.off("change:target");
     model.off("change:fov");
```
Comment (ManonMarchand) [inline js/widget.js:254]: what about turning off shared_target too?

File: src/ipyaladin/__init__.py [hunk 7]
```diff
@@ -1,9 +1,12 @@
 import importlib.metadata
 import pathlib
-from typing import ClassVar
```
Comment (ManonMarchand) [inline src/ipyaladin/__init__.py:3]: remove the change of typing module position (the logic is first block for built-in, second block for external libraries. And in the block follow the alphabetical order)

**Review by ManonMarchand — CHANGES_REQUESTED** [review]

File: src/ipyaladin/__init__.py [hunk 8]
```diff
@@ -105,6 +118,45 @@ def _handle_custom_message(self, model, message, list_of_buffers):  # noqa: ARG0
...
+        """
+        Set the target of the Aladin Lite widget.
+        :param target: string or astropy.coordinates.SkyCoord object
+        :return: None
+        """
+        if isinstance(target, str):  # If the target is a string, parse it
+            sc = parse_coordinate_string(target)
+            self._target = f"{sc.icrs.ra.deg} {sc.icrs.dec.deg}"
+        elif isinstance(target, SkyCoord):  # If the target is a SkyCoord object
+            self._target = f"{target.icrs.ra.deg} {target.icrs.dec.deg}"
```
Comment (ManonMarchand) [inline src/ipyaladin/__init__.py:146]: Could also be moved out of the if/else

**Comment by ManonMarchand** [comment]
Merged, thanks Tom! :tada:

**Pull request merged**